### PR TITLE
change-locale mutation fixes

### PR DESCRIPTION
**Pull request by fjolne**
I tried to understand what's wrong with `force-root-render!` call: `React.render` is actually called and receives updated `data-tree`, but somehow UI just doesn't update. Small timeout helps though. Any ideas?

### CODE REVIEW

File: src/main/com/fulcrologic/fulcro_i18n/i18n.cljc [hunk 1]
```diff
@@ -87,7 +87,7 @@
   (action [{:keys [state app]}]
     (ensure-locale-loaded! app locale)
     (swap! state assoc ::current-locale (comp/get-ident Locale {::locale locale}))
-    (app/force-root-render! app))
+    (js/setTimeout #(app/force-root-render! app) 200))
```
Comment (awkay) [inline src/main/com/fulcrologic/fulcro_i18n/i18n.cljc:90]: Hm. seems like a regression. Not sure why a timeout would help.

**Comment by awkay** [comment]
If this is working, let's just go with it for now. I don't have the extra time to look further into it.

**Comment by awkay** [comment]
I had to fix a mistake you made in the CLJC (use js ns). Please run tests in future (usually just `make` at command line in my projects).  Please test 0.0.4-alpha-SNAPSHOT in your project.

**Comment by fjolne** [comment]
Ah, sorry for tests, blindly assumed there would be none for such a small lib.

`schedule-render!` unfortunately doesn't make it.

**Comment by fjolne** [comment]
Moreover, even transacting `change-locale!` again with the same params doesn't re-render UI. BUT `schedule-render!` works as expected when called outside of mutation.

I recall you made some changes to transactions called from inside mutations to go to the next render tick in order to not block UI for heavy txs, maybe it has something to do with that?

**Comment by fjolne** [comment]
Seems like `schedule-render!` conflicts with the one inside `tx-processing/process-queue!`: if I remove that call then repeating `change-locale!` tx works.

**Comment by fjolne** [comment]
Anyway, it's perfectly fine if you don't have time to dig into it, I'm ok going with fork. My primary motivation is that I don't understand how this works and what implications it can have in unrelated scenarios.

**Comment by awkay** [comment]
So, I suspect that props are not changing (you mention calling it again with same params), which would normally cause S.C.U. to prevent React from updating anything.  The force root option on render sets a dynamic var that is used within the pre-written S.C.U. hook to tell it to answer `true` instead of running the check. It's used mainly on hot code reload (since props have not changed) and for changing locale. No idea why there would be a timing issue.

**Comment by fjolne** [comment]
From what I see, tx processing flow hijacks force-root render scheduling a common render instead. But even without that hijacking timeout is still required... Let me dig into it a bit more, thanks for the explanation about SCU!

**Comment by awkay** [comment]
So, part of the reason this is still marked alpha is the locale stuff in general. I was considering moving the locale and loaded translations  to an external atom so that the library would:

* Be more independent of Fulcro
* The locale rendering could find the current locale and translations even when dynamic vars were not set (many situations where that could be true in async js)
* You really don't need more than one locale per web page, even if you have multiple apps.

I just don't use this lib right now, and have 16 OOS libs I maintain, plus 2 businesses. 

If you wanted to take stuff on more actively, I would be happy to make you a collaborator and you could help make these decisions and such.

**Comment by awkay** [comment]
> From what I see, tx processing flow hijacks force-root render scheduling a common render instead. But even without that hijacking timeout is still required... Let me dig into it a bit more, thanks for the explanation about SCU!

Yeah, I know that that option won't make it through if there is something in queue...just not sure why the explicit call to force isn't working...I'll glance real quick.

**Comment by awkay** [comment]
I see nothing either...the regular (undeferred) call to force-root-render should work.

**Pull request merged**